.cmake/check-format.cmake

```diff
@@ -0,0 +1,31 @@
+set (EXCLUDE_DIR "build/")
+file(GLOB_RECURSE ALL_CXX_SOURCE_FILES *.cpp *.hpp)
+foreach (TMP_PATH ${ALL_CXX_SOURCE_FILES})
+    string (FIND ${TMP_PATH} ${EXCLUDE_DIR} EXCLUDE_DIR_FOUND)
+    if (NOT ${EXCLUDE_DIR_FOUND} EQUAL -1)
+        list (REMOVE_ITEM ALL_CXX_SOURCE_FILES ${TMP_PATH})
+    endif ()
+endforeach(TMP_PATH)
+
+#https://gitlab.cern.ch/Caribou/peary/commit/167da6ffcaca80666f581b1d75e00771fcaeed3a
+add_custom_target(
+    check-format
+    COMMAND ${CLANG_FORMAT} 
+    -style=file 
+    -output-replacements-xml
+    ${ALL_CXX_SOURCE_FILES}
```

Comment:
Instead of the cmake script above, I was wondering whether this would work in cmake (depends if the shell expansion is working or not inside cmake custom target): 
```
add_custom_target(
    check-format
    COMMAND ${CLANG_FORMAT} 
    -style=file 
    -output-replacements-xml
    ${PROJECT_SOURCE_DIR}/*[^build]/**/*.{cpp,hpp}
...
```

Reply:
I tried a similar synthax but it didn't work anyway.